### PR TITLE
SpanArray HTML rendering interactivity and view instance tools

### PR DESCRIPTION
**Pull request by PokkeFe**
For issue #202

Added interactivity to the JS-generated HTML rendering of SpanArrays and TokenSpanArrays, including table controls like visibility toggling and highlight locking. 

Removed JupyterLab-specific CSS overrides in favor of more consistent styling across themes, and made some general styling adjustments.

Added highlights as a view instance property, preserving highlight locks over multiple view renders, but isolating the locks to the instance (Multiple JPL windows of the same notebook can have separate Spans hidden or highlighted at the same time).

Compatibility verified in JupyterLab and VSCode

### CODE REVIEW

File: text_extensions_for_pandas/resources/span_array.js [hunk 1]
```diff
@@ -255,11 +300,151 @@ if(!window.SpanArray || window.SpanArray.VERSION < VERSION) {
...
+            })
+
+            source_spanarray.render()
+        })
+    }
+} else {
+    // SpanArray JS is already defined and not an outdated copy
+    // Replace global SpanArray CSS with latest copy IFF global stylesheet is undefined
+
+    const global_stylesheet = document.head.querySelector("style.span-array-css")
```
Comment (frreiss) [inline text_extensions_for_pandas/resources/span_array.js:441]: This code looks like a duplicate with the block of code at the top of this file. Maybe merge into a single function?

**Review by frreiss — APPROVED** [review]
Some minor suggested tweaks/concerns -- see inline comments -- but overall this is ready to merge.

File: text_extensions_for_pandas/resources/span_array.js [hunk 2]
```diff
@@ -1,7 +1,17 @@
 // Increment the version to invalidate the cached script
-const VERSION = 0.62
+const VERSION = 0.75
 
-if(!window.SpanArray || window.SpanArray.VERSION < VERSION) {
+if(window.SpanArray == undefined || window.SpanArray.VERSION == undefined || window.SpanArray.VERSION < VERSION) {
+
+    // Replace global SpanArray CSS with latest copy
```
[the remark this answers is not in the document]
Comment (PokkeFe) [reply text_extensions_for_pandas/resources/span_array.js:6]: Yes, and this is one of the unfortunate pitfalls of using CSS style blocks. There is no way to declare a block as locally scoped (without JS). With the static backup rendering integrated I'll be able to remove the JS component from global space. One our component is fully local we won't have to worry about handling global stylesheets or fighting with cached scripts.

tl;dr I'm working on a PR to address pitfalls related to versioning and multiple notebook versions.

File: text_extensions_for_pandas/resources/span_array.js [hunk 3]
```diff
@@ -255,11 +300,151 @@ if(!window.SpanArray || window.SpanArray.VERSION < VERSION) {
...
+
+            const closest_tr = event.target.closest("tr")
+            if(closest_tr == undefined) return
+
+            const matching_span = doc_object.lookup_table[closest_tr.getAttribute("data-id")]
+
+            switch(closest_control_button.getAttribute("data-control")) {
+                case "visibility":
+                    {
+                        if(matching_span != undefined) matching_span.visible = !matching_span.visible
```
[the remark this answers is not in the document]
Comment (PokkeFe) [reply text_extensions_for_pandas/resources/span_array.js:392]: Moved the check for an early return.

File: text_extensions_for_pandas/resources/span_array.js (hunk 1 above)
Comment (PokkeFe) [reply text_extensions_for_pandas/resources/span_array.js:441]: Moved the matching constant declarations to outside the block to save some bytes and readability. Both of these global checks will be removed in the PR mentioned in the first comment response once scoping is switched to local and the style block is removed.
